src/dc_mimefactory.rs

```diff
@@ -1023,35 +1003,25 @@ unsafe fn build_body_file(
                 MAILMIME_MECHANISM_BASE64 as libc::c_int,
             );
             if needs_ext {
-                let mut cur1: *mut clistiter = (*(*mime_fields).fld_list).first;
-                while !cur1.is_null() {
-                    let field: *mut mailmime_field = (if !cur1.is_null() {
-                        (*cur1).data
-                    } else {
-                        ptr::null_mut()
-                    }) as *mut mailmime_field;
-                    if !field.is_null()
-                        && (*field).fld_type == MAILMIME_FIELD_DISPOSITION as libc::c_int
+                for cur_data in (*(*mime_fields).fld_list).into_iter() {
```

Comment:
`&clist` implements `IntoIterator`, so it can be written as
```
for cur_data in &(*(*mime_fields).fld_list)
```

Slightly simplier, imho

Reply:
thanks for the note -- there are also many other into_iterator()s in dc_mimeparser -- at this point i don't want to do a global replace but rather look into getting rid more of clists and c code at all.